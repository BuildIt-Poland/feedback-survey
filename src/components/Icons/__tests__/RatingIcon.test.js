import React from 'react';
import { create } from 'react-test-renderer';

import RatingIcon from '../RatingIcon';

describe('COMPONENT - RatingIcon', () => {
  it('does not render if not matching icon', () => {
    const component = create(<RatingIcon iconType="test" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with bad icon', () => {
    const component = create(<RatingIcon iconType="bad" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with average icon', () => {
    const component = create(<RatingIcon iconType="average" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with good icon', () => {
    const component = create(<RatingIcon iconType="good" />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with excellent icon', () => {
    const component = create(<RatingIcon iconType="excellent" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
