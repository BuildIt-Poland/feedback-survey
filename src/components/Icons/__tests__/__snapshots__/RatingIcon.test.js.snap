// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RatingIcon does not render if not matching icon 1`] = `
<svg
  viewBox="0 0 490 490"
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <g>
    <circle
      cx="157.651"
      cy="175.288"
      r="42.642"
    />
    <circle
      cx="332.349"
      cy="175.289"
      r="42.642"
    />
    <path
      d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
    />
  </g>
</svg>
`;

exports[`COMPONENT - RatingIcon renders correctly with average icon 1`] = `
<svg
  viewBox="0 0 490 490"
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <g>
    <path
      d="M137.723,364.15c4.396-3.26,108.787-78.91,214.584,0l18.303-24.554c-124.414-92.81-249.965-0.912-251.221,0.015 L137.723,364.15z"
    />
    <circle
      cx="157.651"
      cy="175.288"
      r="42.642"
    />
    <circle
      cx="332.349"
      cy="175.289"
      r="42.642"
    />
    <path
      d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
    />
  </g>
</svg>
`;

exports[`COMPONENT - RatingIcon renders correctly with bad icon 1`] = `
<svg
  viewBox="0 0 490 490"
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <g>
    <circle
      cx="157.651"
      cy="175.288"
      r="42.642"
    />
    <circle
      cx="332.349"
      cy="175.289"
      r="42.642"
    />
    <path
      d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
    />
  </g>
</svg>
`;

exports[`COMPONENT - RatingIcon renders correctly with excellent icon 1`] = `
<svg
  viewBox="0 0 490 490"
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <g>
    <path
      d="M245,389.513c60.977,0,110.409-49.432,110.409-110.409H134.591C134.591,340.081,184.023,389.513,245,389.513z"
    />
    <circle
      cx="157.651"
      cy="175.288"
      r="42.642"
    />
    <circle
      cx="332.349"
      cy="175.289"
      r="42.642"
    />
    <path
      d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
    />
  </g>
</svg>
`;

exports[`COMPONENT - RatingIcon renders correctly with good icon 1`] = `
<svg
  viewBox="0 0 490 490"
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <g>
    <path
      d="M119.375,325.465c0.837,0.613,56.644,41.466,130.53,41.466c37.205,0,79.015-10.363,120.691-41.452l-18.303-24.554
c-105.857,78.955-210.188,3.26-214.584,0L119.375,325.465z"
    />
    <circle
      cx="157.651"
      cy="175.288"
      r="42.642"
    />
    <circle
      cx="332.349"
      cy="175.289"
      r="42.642"
    />
    <path
      d="M69.086,490h351.829C459.001,490,490,459.001,490,420.914V69.086C490,30.999,459.001,0,420.914,0H69.086 C30.991,0,0,30.999,0,69.086v351.829C0,459.001,30.991,490,69.086,490z M30.625,69.086c0-21.204,17.257-38.461,38.461-38.461 h351.829c21.204,0,38.461,17.257,38.461,38.461v351.829c0,21.204-17.257,38.461-38.461,38.461H69.086 c-21.204,0-38.461-17.257-38.461-38.461V69.086z"
    />
  </g>
</svg>
`;
